src/player.cpp

```diff
@@ -9610,32 +9610,41 @@ bool player::wield( item& target )
 
     int mv = 0;
 
+    bool target_removed = false;
+    item target_copy = target;
     if( is_armed() ) {
+        inventory old_inv = inv;
```

Comment:
Copying the whole inventory (and then assigning it back upon a condition) looks like a rather dirty hack.